perfkitbenchmarker/providers/gcp/gce_network.py

```diff
@@ -234,28 +241,150 @@ class GceNetwork(network.BaseNetwork):
+                                               self.cidr, self.project)
+
+    # Stage FW rules.
+    self.all_nets = self._GetNetworksFromSpec(
+        network_spec)  # Holds the different networks in this run.
+    self.external_nets_rules = {}  # Holds FW rules for any external subnets.
+
+    #  Set the default rule to allow all traffic within this network's subnet.
+    firewall_name = self._MakeGceFWRuleName()
     self.default_firewall_rule = GceFirewallRule(
-        firewall_name, self.project, ALLOW_ALL, name, NETWORK_RANGE)
+        firewall_name, self.project, ALLOW_ALL, name, self.cidr)
+
+    # Set external rules to allow traffic from other subnets in this benchmark.
+    for ext_net in self.all_nets:
+      if ext_net == self.cidr:
+        continue  # We've already added our own network to the default rule.
+      rule_name = self._MakeGceFWRuleName(dst_cidr=ext_net)
+      self.external_nets_rules[rule_name] = GceFirewallRule(rule_name,
+                                                            self.project,
+                                                            ALLOW_ALL, name,
+                                                            ext_net)
+
+  def _GetNetworksFromSpec(self, network_spec):
+    """Returns a list of distinct CIDR networks for this benchmark.
+
+    All GCP networks that aren't set explicitly with a vm_group cidr property
+    are assigned to the default network. The default network is either
+    specified as a single region with the gce_subnet_region and
+    gce_subnet_addr flags, or assigned to an auto created /20 in each region
+    if no flags are passed.
+
+    Args:
+      network_spec: The network spec for the network.
+    Returns:
+      A set of CIDR strings used by this benchmark.
+    """
+    nets = set()
+    gce_default_subnet = (FLAGS.gce_subnet_addr if FLAGS.gce_subnet_region
+                          else NETWORK_RANGE)
+
+    if network_spec.custom_subnets:
+      for (k, v) in network_spec.custom_subnets.items():
+        if not v['cidr'] and not v['cloud'] == 'GCP':
+          pass  # @TODO handle other providers defaults in net_util
+        elif not v['cidr']:
+          nets.add(gce_default_subnet)
+        else:
+          nets.add(v['cidr'])
+    return nets
+
+  def _MakeGceNetworkName(self, net_type=None, cidr=None, uri=None):
+    """Build the current network's name string.
+
+    Uses current instance properties if none provided.
+    Must match regex: r'(?:[a-z](?:[-a-z0-9]{0,61}[a-z0-9])?)'
+
+    Args:
+      net_type: One of ['default', 'single', 'multi']
+      cidr: The CIDR range of this network.
+      uri: A network suffix (if different than FLAGS.run_uri)
+    Returns:
+      String The name of this network.
+    """
+    if FLAGS.gce_network_name:  # Return user managed network name if defined.
+      return FLAGS.gce_network_name
+
+    _net_type = self.net_type if not net_type else net_type
```

Comment:
please be consistent with style, remove starting underscore from local variables.
also these can be rewritten as:
_net_type = self.net_type if not net_type else net_type
net_type = net_type or self.net_type

Reply:
Fixed.